worker/expiration_worker.go

```diff
@@ -44,37 +42,60 @@ type ExpirationWorker struct {
 }
 
 // GetExpirationWorker returns a new scores expirer worker
-func GetExpirationWorker(configPath string, logger zap.Logger) (*ExpirationWorker, error) {
+func GetExpirationWorker(configPath string) (*ExpirationWorker, error) {
 	worker := &ExpirationWorker{
 		ConfigPath: configPath,
-		Logger:     logger,
 		Config:     viper.New(),
 	}
-	err := worker.configure()
+
+	err := worker.loadConfiguration()
 	if err != nil {
 		return nil, err
 	}
+
+	err = worker.configure()
+	if err != nil {
+		return nil, err
+	}
+
 	return worker, nil
 }
 
-func (w *ExpirationWorker) configure() error {
-	err := w.loadConfiguration()
+// GetExpirationWorkerExtCfg returns a new scores expirer worker with already loaded configuration.
+func GetExpirationWorkerExtCfg(host string, port int, password string, db int, connectionTimeout int,
```

Comment:
I think `NewExpirationWorker` would be a better name.

Reply:
Makes sense. Changing it.